Automattic/wp-calypso
# Theme showcase: Add specific URL for "All" and "My Themes" categories

Follow-up of https://github.com/Automattic/wp-calypso/pull/79929

## Proposed Changes

Creates specific paths for the "All" and "My Themes" categories in the theme showcase, so users can go back to them after selecting a single theme.

## Testing Instructions

- Use the Calypso live link below.
- Search for a theme.
- Make sure the "All" category is automatically selected.
- Check the URL has changed to `/themes/all/:site`.
- Select a theme.
- Click on the "Back to themes" button.
- Make sure you land in the "All" category.
- Reload the `/themes/all/:site` URL.
- Make sure the "All" category is automatically selected.
- Switch to a Jetpack/Atomic site.
- Activate the "My Themes" category.
- Check the URL has changed to `/themes/my-themes/:site`.
- Select a theme.
- Click on the "Back to themes" button.
- Make sure you land in the "My Themes" category.
- Reload the `/themes/my-themes/:site` URL.
- Make sure the "My Themes" category is automatically selected.
- Test any other scenario you can think of (e.g. filter by feature, switch to a different category, etc) and make sure there are no regressions.
